build_barback/lib/src/transformer/transformer.dart

```diff
@@ -48,10 +48,14 @@ class BuilderTransformer
         (id) => transform
             .readInputAsString(toBarbackAssetId(id))
             .catchError((_) => null));
-    return Future.wait(inputs.map((input) => _apply(input, transform)));
+    var resourceManager = new ResourceManager();
+    return Future
+        .wait(inputs.map((input) => _apply(input, transform, resourceManager)))
+        .then((_) => resourceManager.disposeAll());
```

Comment:
Do we want to handle the failure case too or are we guaranteed there is no error here? `whenComplete` will fire regardless of whether it completes with a result or an error.